gpii/node_modules/contextManager/src/ContextManagerUtilities.js

```diff
@@ -95,11 +97,79 @@ gpii.contextManager.utils.findActiveContexts = function (currentContext, matchDa
         var result = fluid.model.transformWithRules(currentContext, { evaluations: condArray });
         // if we only have 'true' entries after evaluation
         if (result.evaluations && $.inArray(false, result.evaluations) === -1) {
-            // TODO Take priorities into account
+            // TODO  (done->needs checking)Take priorities into account
+			if (activeContexts["indices"].length < 1){
+				activeContexts["indices"].push(contextId);
+				activeContexts["saliences"].push(content["salience"]);
+				
+			} else {
+				var pos = 0;
+				while ((activeContexts["saliences"][pos])&&(activeContexts["saliences"][pos] > content["salience"])){
+					pos++;
+				}
+				activeContexts["indices"].splice(pos,0,contextId);
+				activeContexts["saliences"].splice(pos,0,content["salience"]);
+				
+				
+			};
+			
             // prepend the entry
-            activeContexts.unshift(contextId);
+            // activeContexts.unshift(contextId);
         }
     });
-    fluid.log("Active contexts calculated to be: " + activeContexts);
-    return activeContexts;
-};
\ No newline at end of file
+	// if there are no feasible contexts, then choose gpii-default as the active one
+	// note that this breaks Kasper's UtilitiesTests as they always expect to have gpii-default active
+	if (activeContexts["indices"].length < 1){
+		activeContexts["indices"] = [ "gpii-default" ];
+	}
+    fluid.log("Active contexts calculated to be: " + activeContexts["indices"]);
+    return activeContexts["indices"];
+};
+
+
+
+
+gpii.contextManager.utils.expandContext = function (userToken, matchData){
+	var newMatchData = fluid.copy(matchData);
+	var i = 0;
+	// TODO: (done->needs checking) add priorities if they are not already added
+	fluid.each( newMatchData.inferredConfiguration , function ( inferredConfiguration ) {
+		// fluid.log("inferredConfiguration is "+ JSON.stringify (inferredConfiguration, null, " "));
+		
+		if (inferredConfiguration.salience === undefined){
+		// then it is user-stated, should be ge 1024
+			inferredConfiguration.salience = 1024+i;
+			i++;
+			
+		}
+	
+	});
+	// TODO: (done->needs checking) WiP add contexts
+	if (userToken === "sammy"){
```

Comment:
Is there a motivation for adding this here, as opposed to the NP set? We try to hardcode as little as possible, and while the conditions of the NP set are a type of hardcoding as well, it's a lot more flexible to have things like this in the NP-set as opposed to the codebase